BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveCore
+
+/// A store contains data for swap tokens
+public class SwapTokenStore: ObservableObject {
+  /// All  tokens
+  @Published var allTokens: [BraveWallet.ERCToken] = []
+  /// The current selected token to swap from. Default with nil value.
+  @Published var selectedFromToken: BraveWallet.ERCToken? {
+    didSet {
+      if let token = selectedFromToken {
+        fetchTokenBalance(for: token) { [self] balance in
+          selectedFromTokenBalance = balance
```

Comment:
May want to do `[weak self]` in cases where we plan to eventually destroy the store when the user leaves the screen and we dont care about the result anymore